server/tests/test_notebook_view.py

```diff
@@ -149,6 +149,36 @@ def test_tryit_view(client, fake_user, logged_in, iomd):
         assert len(response.redirect_chain) == 0
 
 
+@pytest.mark.parametrize("logged_in", [True, False])
+@pytest.mark.parametrize("request_method", ["POST", "GET"])
+def test_tryit_view_file_uploads(client, fake_user, logged_in, request_method):
+    import tempfile
+    import mimetypes
+
+    # we should see the same script blocks
+    # for both logged_in and both request_method
+
+    with tempfile.NamedTemporaryFile(mode="w+") as f:
+        f.name = "data.txt"
+        file_content = "testing"
+        file_mimetype = mimetypes.guess_type(f.name)
+        f.write(file_content)
+        f.seek(0)
+        path = reverse("try-it")
+
+        def http_response(request_method):
+            if request_method == "POST":
+                return client.post(path, {"filename": f.name, "file": f}, follow=True)
+            else:
+                return client.get(path, {"filename": f.name, "file": open(f.name)}, follow=True)
```

Comment:
Try just encoding this in the URL:

client.get(f'/tryit?file={file_content}&filename={filename}')